extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/devmode/UndertowHotReplacementSetup.java

```diff
@@ -10,14 +10,16 @@
 import io.quarkus.undertow.runtime.UndertowDeploymentRecorder;
 
 public class UndertowHotReplacementSetup implements HotReplacementSetup {
-
+    // TODO: This is marked as protected but I don't see anyone else (other than this class) using it.
+    // get rid of this if not used anywhere else or at least make it private
     protected static final String META_INF_SERVICES = "META-INF/resources";
 
     @Override
     public void setupHotDeployment(HotReplacementContext context) {
         List<Path> resources = new ArrayList<>();
         for (Path i : context.getResourcesDir()) {
-            Path resolved = i.resolve(META_INF_SERVICES);
+            final String nameSeparator = i.getFileSystem().getSeparator();
```

Comment:
I agree this is theoretically more correct but we use `File.separator` everywhere in Quarkus so we could use that in the constant instead.

Reply:
That (existing) constant actually is getting used in multiple different semantical ways - one for filesystem resource resolution and the other is to load resources from a classloader (which mandates the usage of `/` irrespective of the runtime). I can create a new constant solely for this purpose and then set its value to use `File.separator`. However, the reason I decided to go this approach is, I'm not yet 100% sure how the existing tests (one of which Stuart added recently exactly for this issue) are passing on Windows OS. Perhaps those test setup have mounted filesystem whose name separator is `/` ? That's why I decided to use the name separator that is more relevant/dynamic based on the `Path`'s `FileSystem` rather than the runtime OS. I'm open to changing this, if you think we should use `File.separator`.

Reply:
The existing tests are not run on windows. The native image tests are only run on Linux

Reply:
OK, so that makes sense then. @stuartwdouglas, @gsmet do you want me to use `File.separator` instead of this current change? Let me know, I will update this PR.